api_app/analyzers_manager/observable_analyzers/dns/dns_malicious_detectors/mullvad_dns.py

```diff
@@ -0,0 +1,127 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+
+import base64
+import logging
+from urllib.parse import urlparse
+
+import dns.message
+import httpx
+
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+from api_app.analyzers_manager.exceptions import AnalyzerConfigurationException
+from api_app.analyzers_manager.observable_analyzers.dns.dns_responses import (
+    malicious_detector_response,
+)
+from api_app.choices import Classification
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class MullvadDNSAnalyzer(ObservableAnalyzer):
+    """
+    MullvadDNSAnalyzer:
+
+    This analyzer queries Mullvad's DNS-over-HTTPS service (using the "base" endpoint)
+    to check a domain's DNS records. It supports two modes:
+      - "query": returns raw DNS answer data.
+      - "malicious": interprets an NXDOMAIN (rcode==3) as the domain being blocked (i.e., malicious).
+    """
+
+    url = "https://base.dns.mullvad.net/dns-query"
+    mode: str = "query"
+
+    def update(self):
+        pass
+
+    @staticmethod
+    def encode_query(observable: str) -> str:
+        """
+        Constructs a DNS query for the given observable (domain) for an A record,
+        converts it to wire format, and encodes it in URL-safe base64.
+        """
+        logger.info(f"Encoding DNS query for {observable}")
+        query = dns.message.make_query(observable, dns.rdatatype.A)
+        wire_query = query.to_wire()
+        encoded_query = (
+            base64.urlsafe_b64encode(wire_query).rstrip(b"=").decode("ascii")
+        )
+        logger.info(f"Encoded query: {encoded_query}")

```

Comment:
for each log, it is important to keep track of the original observable, otherwise there is no way to distinguish one analyzer execution from another